unified-runtime/source/adapters/level_zero/v2/memory.cpp

```diff
@@ -113,18 +136,14 @@ void *ur_integrated_buffer_handle_t::getDevicePtr(
 void *ur_integrated_buffer_handle_t::mapHostPtr(
     ur_map_flags_t /*flags*/, size_t offset, size_t /*size*/,
     ze_command_list_handle_t /*cmdList*/, wait_list_view & /*waitListView*/) {
-  // TODO: if writeBackPtr is set, we should map to that pointer
-  // because that's what SYCL expects, SYCL will attempt to call free
-  // on the resulting pointer leading to double free with the current
-  // implementation. Investigate the SYCL implementation.
+  // For integrated devices, both device and host access the same memory
   return ur_cast<char *>(ptr.get()) + offset;
 }
 
 void ur_integrated_buffer_handle_t::unmapHostPtr(
     void * /*pMappedPtr*/, ze_command_list_handle_t /*cmdList*/,
     wait_list_view & /*waitListView*/) {
-  // TODO: if writeBackPtr is set, we should copy the data back
-  /* nop */
+  // No-op: integrated buffers use zero-copy, no synchronization needed
```

Comment:
what if we go in this path when creating the buffer:
"// Import failed - allocate backing buffer and set up copy-back"
?